### DISCUSSION TITLE
Can a section be skipped by Trees::save, but not by Trees::segmentCloud?

### DISCUSSION THREAD

**Discussion by MadPadCF** [csiro-robotics/raycloudtools#26] [General]
I've been comparing the segmented pointcloud output of Trees::segmentCloud with the txt file output of Trees::save, using implementations of convert_int_to_colour and convert_colour_to_int. 

I've noticed that in many txt file outputs, there are 1-2 missing section ids in the root sections: other than those, all of the root sections have contiguous ids. I initially assumed that these segments had no children, and were filtered out as observed in the Trees::save logic.

However, when I map the point cloud's RGB values back through convert_colour_to_int, I find that those missing segment ids are included; this seems odd, because Trees::segmentCloud also skips segments without children. In the base functionality of RCT this potential discrepancy is not an issue if you use either the cloud outputs or the txt files in isolation, but as I'm attempting to use both, this causes an issue for me; as it means all the section_ids after those missing values are no longer aligned with the RGB values in the point cloud.

Are these values intended to align, and if so, is it intended behaviour that there are RGB colour values with no corresponding section ids?

Many thanks.

**Comment by thomas-lowe** [comment]
Thanks for finding this, I believe the problem is that the root wasn't using the contiguous_section_ids list in Trees::save, so I have pushed this fix to main. 

I didn't have an example of this, but the change hasn't broken my example either. Let me know if this fixes the problem on your data set.

**Reply by MadPadCF** [reply]
Hi Thomas, this change has fixed my issue. Thanks for the quick reply!

**Comment by thomas-lowe** [comment]
[like]  Lowe, Tom (Data61, Pullenvale) reacted to your message:
________________________________
